Library/Homebrew/cmd/tap-info.rb

```diff
@@ -0,0 +1,44 @@
+require "cmd/tap"
+
+module Homebrew
+  def tap_info
+    if ARGV.include? "--all"
+      taps = Tap
+    else
+      taps = ARGV.named.map do |name|
+        tap = Tap.new *tap_args(name)
+        odie "No such tap!" unless tap.installed?
+        tap
+      end
+    end
+
+    if ARGV.json == "v1"
+      print_tap_json(taps)
+    elsif taps.none?
+      files = []
+      count = 0
+      Tap.each do |tap|
+        count +=1
+        files += tap.formula_files
+      end
+      puts "#{count} tap#{plural(count)}, #{files.length} formula#{plural(files.length, "e")}, #{Tap::TAP_DIRECTORY.abv}"
+    else
+      print_tap_info(taps)
+    end
+  end
+
+  private
+
+  def print_tap_info(taps)
+    taps.each_with_index do |tap, i|
+      puts unless i == 0
+      files = tap.formula_files
+      puts "#{tap}: #{files.length} formula#{plural(files.length, "e")} (#{tap.path.abv})"
```

Comment:
Maybe also worth listing the number of commands for e.g. stuff like `homebrew/bundle`?